File: Airwallex/ApplePay/Internal/AWXPlaceDetails+PKContact.m
//
//  AWXPlaceDetails+PKContact.m
//  ApplePay
//
//  Created by Jin Wang on 13/4/2022.
//  Copyright © 2022 Airwallex. All rights reserved.
//

#import "AWXPlaceDetails+PKContact.h"

@implementation AWXPlaceDetails (PKContact)

- (PKContact *)convertToPaymentContact
{
    PKContact *contact = [PKContact new];
    
    NSPersonNameComponents *nameComponents = [NSPersonNameComponents new];
    nameComponents.givenName = self.firstName;
    nameComponents.familyName = self.lastName;
    
    CNMutablePostalAddress *address = [CNMutablePostalAddress new];
    address.ISOCountryCode = self.address.countryCode;
    address.city = self.address.city;
    address.street = self.address.street;
    
    if (self.address.state) {
        address.state = self.address.state;
    }
    
    if (self.address.postcode) {
        address.postalCode = self.address.postcode;
    }
    
    contact.name = nameComponents;
    contact.emailAddress = self.email;
    contact.phoneNumber = [CNPhoneNumber phoneNumberWithStringValue:self.phoneNumber];
    contact.postalAddress = address;
    
    return contact;
}

@end

File: Airwallex/ApplePay/Internal/AWXPlaceDetails+PKcontact.swift
//
//  AWXPlaceDetails+PKcontact.swift
//  ApplePay
//
//  Created by Tony He (CTR) on 2024/7/29.
//  Copyright © 2024 Airwallex. All rights reserved.
//

import Foundation

@objc extension AWXPlaceDetails {
    
    public func convertToPaymentContact() -> PKContact {
        let contact = PKContact()
        var nameComponents = PersonNameComponents()
        nameComponents.givenName = firstName
        nameComponents.familyName = lastName
        
        let addr = CNMutablePostalAddress()
        if let city = address?.city {
            addr.city = city
        }
        if let street = address?.street {
            addr.street = street
        }
        if let countryCode = address?.countryCode {
            addr.isoCountryCode = countryCode
        }
        if let state = address?.state {
            addr.state = state
        }
        if let postcode = address?.postcode {
            addr.postalCode = postcode
        }
        
        contact.name = nameComponents
        if let email = email {
            contact.emailAddress = email
        }
        if let phoneNumber = phoneNumber {
            contact.phoneNumber = CNPhoneNumber(stringValue: phoneNumber)
        }
        contact.postalAddress = addr
        
        return contact
    }
    
}
